tools/circt-bmc/circt-bmc.cpp

```diff
@@ -264,8 +264,10 @@ static LogicalResult executeBMC(MLIRContext &context) {
     SmallVector<StringRef, 4> sharedLibraries(sharedLibs.begin(),
                                               sharedLibs.end());
     mlir::ExecutionEngineOptions engineOptions;
-    engineOptions.transformer = mlir::makeOptimizingTransformer(
-        /*optLevel*/ 3, /*sizeLevel=*/0, /*targetMachine=*/nullptr);
+    std::function<llvm::Error(llvm::Module *)> transformer =
```

Comment:
Don't you have to declare the variable outside this nested scope, because otherwise it'd be deallocated at the `engine->invokePacked` call, right? (also in `circt-lec.cpp`)